### DISCUSSION TITLE
Lucid nullable HasOne relationship Database

### DISCUSSION THREAD

**Discussion by muniter** [adonisjs/core#4604] [Help]
When declaring a model property as:

```typescript
Person extends BaseModel {
...
@hasOne(() => Address)
address: HasOne<typeof Address> | null
...
}
```

Type inference breaks in: 

```typescript
Person.query().preload('address')
```

With error `Argument of type 'address' is not assignable to parameter of type 'undefined'.` If I remove the null from the type definition of the property the preloading works correctly.

I believe this functionality is allowed as discribed in this PR: https://github.com/adonisjs/lucid/pull/944

Anyone knows what's happening?

**Comment by JAWS-tm** [comment]
I’m facing the same issue. Did you solve it?
